src/Symfony/Component/Security/Core/Encryption/SymmetricEncryption.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Encryption;
```

Comment:
This class is only used in `security-http`. What's about marking it internal, and moving it in `Securty\Http`  namespace? In that way, we can also revert  `symfony/secrutiy-core: ^5.3` requirement.

Reply:
Yeah. But I really like this class. :)

I believe this is really useful outside the symfony namespace. Don’t you wake up every day and think about data you want to encrypt?

A follow up PR might even add asymmetric encryption. 

I didn’t want to create a new component and they don’t really belong in the http namespace.

Reply:
Until we have an interface for this service (which could even push us to add `Sodium` to the classname/namespace), I'd suggest to make it internal.

Reply:
I agree with Tobias here, but let's add an interface which could be experimental in the first step. Afterwards we can change the signature of the LoginLink class to receive the interface instead of depending on the implementation.

Reply:
I want to insist. I've added an experimental tag and an interface.